sdks/python/apache_beam/examples/inference/gemini_image_generation.py

```diff
@@ -0,0 +1,190 @@
+      type=str,
+      required=False,
+      help='Gemini Developer API key.')
+  parser.add_argument(
+      '--cloud_project',
+      dest='project',
+      type=str,
+      required=False,
+      help='GCP Project')
+  parser.add_argument(
+      '--cloud_region',
+      dest='location',
+      type=str,
+      required=False,
+      help='GCP location for the Endpoint')
+  return parser.parse_known_args(argv)
+
+
+class PostProcessor(beam.DoFn):
+  def process(self, element: PredictionResult) -> Iterable[Image.Image]:
+    try:
+      response = element.inference
+      for part in response.parts:
+        if part.text is not None:
+          print(part.text)
+        elif part.inline_data is not None:
+          image = Image.open(BytesIO(part.inline_data.data))
+          yield image
+    except Exception as e:
+      print(f"Can't decode inference for element: {element.example}, got {e}")
+      raise e
+
+
+class _WriteImageFn(beam.DoFn):
+  """
+    A DoFn that writes a single PIL.Image.Image object to a file.
+
+    This DoFn is intended to be used internally by the WriteImages transform.
+  """
+  def __init__(self, output_dir: str, filename_prefix: str):
+    """
+        Initializes the _WriteImageFn.
+
+        Args:
+            output_dir: The directory where the image files will be written.
+            filename_prefix: A prefix to use for the output filenames. A UUID
+                and the .png extension will be appended to this prefix.
+    """
+    self._output_dir = output_dir
+    self._filename_prefix = filename_prefix
+
+  def setup(self):
+    """
+        Ensures the output directory exists.
+    """
+    if not FileSystems().exists(self._output_dir):
+      FileSystems().mkdirs(self._output_dir)
+
+  def process(self, image: Image.Image):
+    """
+        Saves the given PIL Image to a PNG file.
+
+        Args:
+            image: A PIL.Image.Image object.
+    """
+    # Generate a unique filename to prevent collisions.
+    unique_id = uuid.uuid4()
+    filename = f"{self._filename_prefix}-{unique_id}.png"
+    output_path = os.path.join(self._output_dir, filename)
+
+    try:
+      logging.debug("Writing image to %s", output_path)
+      with FileSystems().create(output_path) as image_file:
+        image.save(image_file, "PNG")
+    except Exception as e:
+      logging.error("Failed to write image to %s: %s", output_path, e)
+      raise
+
+
+class WriteImages(beam.PTransform):
+  """
+    An Apache Beam PTransform for writing a PCollection of PIL Image objects
+    to individual PNG files in a specified directory.
+  """
+  def __init__(self, output_dir: str, filename_prefix: Optional[str] = "image"):
+    self._output_dir = output_dir
+    self._filename_prefix = filename_prefix
+
+  def expand(self, pcoll: beam.PCollection) -> beam.pvalue.PDone:
+    return (
+        pcoll
+        | "WriteImageToFile" >> beam.ParDo(
+            _WriteImageFn(self._output_dir, self._filename_prefix)))
```

Comment:
We should probably use https://beam.apache.org/releases/pydoc/current/apache_beam.io.fileio.html#apache_beam.io.fileio.WriteToFiles